### ISSUE TITLE
extend_read should not shrink genes.

### ISSUE THREAD

**Issue by ericfournier2** [ArnaudDroitLab/metagene2#10]
If the extend_read value is smaller than certain reads, those should not be affected. Right now, those are shrunk.

An appropriate test is commented out at the end of the test.metagene_extend_reads in inst/unitTests/test_metagene.R . Make sure to uncomment it once this bug is fixed.

**Comment by ericfournier2** [comment]
Unit test now stand-alone in test.metagene_extend_reads_no_shrink

**Comment by ericfournier2** [comment]
We've received a request from a metagene(1) user inquiring about how to trim reads. extend_reads' current implementation happened to fit his needs.

> In our specific use case, we are interested in only the 5’ position of the read, and we are dealing with reads that have variable length. As such, your tool is working well for us, but we would like to only display either:
> 
> 1) Only the 5’ mapping position of each read
> 
> or
>
> 2) Only a specific number of bases for each read
> 
> In option 1, I think this would require adding a new parameter option within your code. This is something that is important for many assays which are fine-mapping genomic events. We would be very grateful for this additional utility.
> 
> For option 2, we can use a tool like BamUtil’s trimBam to soft-clip the already mapped reads to a shorter length. For this option, we would like to know if your BAM read counter is still going to calculate coverage over a genomic position if the read has soft-clipped bases at that position.

After advising them of the existence of this new parameter in metagene2:
> We’ll fix read lengths to something short (5bp) for more fine-mapped visualization purposes. We are essentially using this tool to display meta-plots of transposon insertion events into the genome. 

So maybe what we need is a "trim_reads" and an "extend_reads" option. Maybe a "set_read_size" one? Shrinking reads are still a bug given the parameter's name, but the functionality should be kept.

So, basically, this bug covers a needed functionality. They ended up using 5bp as the extend_reads parameter
